cm/README.md

```diff
@@ -18,14 +11,202 @@ It must contain:
 * A setup.cfg symlink to the cm/setup.cfg file. (see
   https://mokacoding.com/blog/symliks-in-git/ for a brief explanation on symbolic links)
 * A worker.py entrypoint, this must be an executable that start the worker.
-* A test.py unittest testing the calculation without service dependencies.
-* A schema.json jsonschema which describes the schema used for the input. You can find additional information about jsonschema
-  On https://json-schema.org/.
+* A test.py unittest testing the calculation without service dependencies (see 
+  https://docs.python.org/3/library/unittest.html for explination about unittest library).
+* A schema.json jsonschema which describes the schema used for the input. 
+  You can find additional information about jsonschema
+  (see https://json-schema.org/).
+* A requirements.txt including cm specific dependencies.
+* A Dockerfile to create the necessary docker image.
+  The docker-compose will need to be modified accordingly to create the new image.
+* A python script relating to the cm code.
+* A readme file describing the cm operation.
+
+Below is an example of the tree structure :
+
+```bash
+─── cm
+    └── first_cm
+        ├── setup.cfg
+        ├── worker.py
+        ├── test.py
+        ├── schema.json
+        ├── requirements.txt
+        ├── Dockerfile
+        ├── new_cm.py
+        └── README.md
+```
+
+
+## Setup.cfg
+
+Below is the setup file configuration.
+
+```
+[flake8]
+max-line-length=88
+[isort]
+line_length=88
+```
+
+To keep consistency in the project, 
+the following parameters mustn't be changed.
+
+## Worker.py
+
+Below is an example of the broker implementation.
+
+````python
+#!/usr/bin/env python3
+import BaseCM.cm_base as cm_base
+
+from new_cm import process
+
+
+app = cm_base.get_default_app()
+schema_path = cm_base.get_default_schema_path()
+
+
+@app.task(base=cm_base.CMBase, bind=True, schema_path=schema_path)
+def fun(self, selection: dict, rasters: list, params: dict):
+    """New cm description"""
+    result = process(selection, rasters, params)
+    return result
+
+
+if __name__ == "__main__":
+    cm_base.start_app(app)
+````
+
+The selected area and the selected raster 
+are in the variable ```selection``` and ```rasters``` respectively.
+
+The variable ```params``` refers to the data provided by the form on the frontend.
+
+## Test.py
+
+Below is an example of cm test.
+
+```python
+import unittest
+
+from new_cm import process
+
+selection = "selection.shp"
+raster = "raster.tif"
+params = 10
+
+class TestNewCM(unittest.TestCase):
+    def test_new_cm_feature(self):
+        result = process(selection, rasters, params)
+        self.assertIsNotNone(result)
+
+
+if __name__ == "__main__":
+    unittest.main()
+```
+
+## Schema.json
+
+Below is an example of the data schema expected by the cm.
```

Comment:
Specify that this is referring to the optional input parameter

Reply:
I've made the change